core/src/main/java/kafka/server/SharePartition.java

```diff
@@ -221,9 +224,18 @@ public static RecordState forId(byte id) {
      * The state epoch is used to track the version of the state of the share partition.
      */
     private int stateEpoch;
+    /**
+     * The replica manager is used to get the earliest offset of the share partition, so we can adjust the start offset.
+     */
+    private final ReplicaManager replicaManager;
+    /**
+     * The pending write state batches are used to track the state batches that are pending to be written to persister.
+     * They originate from the archiving of records due to LSO advancing forward.
+     */
+    private final List<PersisterStateBatch> pendingWriteStateBatches;
```

Comment:
Do we require it or just move the SPSO in next acknowledgement so previous batches will anywyas be voided by persister?

Reply:
we will be removing it as disscused on the conversation thread https://confluent.slack.com/archives/C06E4G1KHDJ/p1716547825702809